robot-core/src/main/java/org/obolibrary/robot/ReportOperation.java

```diff
@@ -610,29 +612,42 @@ public static boolean processReport(Report report, String outputPath, Map<String
   }
 
   /**
-   * Given a set of rules (either as the default rule names or URL to a file), return a map of the
-   * rule names and the corresponding query strings.
+   * Create a list of ReportQuery objects based on the profile.
    *
-   * @param rules set of rules to get queries for
-   * @return map of rule name and query string
-   * @throws IOException on any issue reading the query file
-   * @throws URISyntaxException on issue converting URL to URI
+   * @param profile Map of rule name to violation level
+   * @return List of Report Query objects
+   * @throws IOException on problem getting query strings
+   * @throws URISyntaxException on problem getting query strings
    */
-  private static Map<String, String> getQueryStrings(Set<String> rules)
+  private static Set<ReportQuery> getReportQueries(Map<String, String> profile)
```

Comment:
Do you think it would be better to preserve the order of the queries that the user provides, or not?

Reply:
We didn't in the past, but we can switch this to be a `LinkedHashMap` which should preserve the order

Reply:
I was thinking a List.

Sorry to be picky. I'm quite happy with this PR now.